app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedFragment.java

```diff
@@ -0,0 +1,216 @@
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+import java.util.Objects;
+
+import ru.vssemikoz.newsfeed.R;
+import ru.vssemikoz.newsfeed.adapters.NewsFeedAdapter;
+import ru.vssemikoz.newsfeed.dialogs.PickCategoryDialog;
+import ru.vssemikoz.newsfeed.models.Category;
+import ru.vssemikoz.newsfeed.models.NewsItem;
+import ru.vssemikoz.newsfeed.navigator.Navigator;
+import ru.vssemikoz.newsfeed.storage.IconicStorage;
+
+import static androidx.core.util.Preconditions.checkNotNull;
+
+public class NewsFeedFragment extends Fragment implements NewsFeedContract.View,
+        PickCategoryDialog.OnCategorySelectedListener {
+    private String TAG = NewsFeedFragment.class.getName();
+    private String CURRENT_CATEGORY = "CURRENT_CATEGORY";
+    private String CURRENT_SHOW_FAVORITE = "CURRENT_SHOW_FAVORITE";
+
+    private NewsFeedContract.Presenter presenter;
+    private Context context;
+    private NewsFeedAdapter adapter;
+    private RecyclerView recyclerView;
+    private TextView emptyView;
+    private ImageButton favoriteNewsButton;
+    private ImageButton categoryButton;
+    private ProgressBar progressBar;
+    private TextView descriptionView;
+
+    public NewsFeedFragment() {
+    }
+
+    static NewsFeedFragment newInstance() {
+        return new NewsFeedFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        Log.d(TAG, "onCreate: ");
+        super.onCreate(savedInstanceState);
+
+        context = Objects.requireNonNull(getActivity()).getApplicationContext();
+        adapter = new NewsFeedAdapter(context);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        updateCategoryNameOnDescription(Category.getDisplayName(Category.ALL));
+        presenter.start();
+    }
+
+    @Override
+    public void setPresenter(NewsFeedContract.Presenter presenter) {
+        Log.d(TAG, "setPresenter: ");
+        this.presenter = checkNotNull(presenter);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.newsfeed_frag, container, false);
+
+        recyclerView = root.findViewById(R.id.rv_news_feed);
+        emptyView = root.findViewById(R.id.tv_empty_view);
+        favoriteNewsButton = root.findViewById(R.id.ib_favorite);
+        progressBar = root.findViewById(R.id.progress_bar);
+        descriptionView = root.findViewById(R.id.tv_description);
+        categoryButton = root.findViewById(R.id.ib_category);
+        initRecyclerView();
+        favoriteNewsButton.setOnClickListener(v -> {
+            presenter.invertFavoriteState();
+
+        });
+        categoryButton.setOnClickListener(v -> {
+            Navigator.openCategoryDialogFragment(this, Objects.requireNonNull(getActivity()));
```

Comment:
This should call a presenter which will call a corresponding `View` method to show a dialog.